src/modules/shared/forms/components/checkbox-array.component.ts

```diff
@@ -103,31 +108,47 @@ export class FormCheckboxArrayComponent implements OnInit, DoCheck {
         this.conditionalFormControl(item.conditional.id).updateValueAndValidity();
 
       }
-
+      
     });
-
+    
     this.cdr.detectChanges();
-
+    
   }
-
-
+  
+  
   isChecked(value: string): boolean {
     return this.fieldArrayValues.includes(value);
   }
-
+  
   onChanged(e: Event): void {
-
+    
     const event = e.target as HTMLInputElement;
     const valueIndex = (this.fieldArrayControl.value as string[]).indexOf(event.value);
-
+    
     if (event.checked && valueIndex === -1) {
+      
+      (this.isItemExclusive(event.value) || this.isExclusiveChecked()) && this.fieldArrayControl.clear(); 

```

Comment:
I would prefer if we keep using the if, is more verbose but at the same time is more readable! 

Then I think you just have to check if is the exclusive item, the second condition seems unnecessary.

Reply:
I agree with Diogo! It was a bit hard for me to read and understand what should happen

Reply:
Ok I will change it to an 'if'!
The difference is one checks if the 'clicked' item is exclusive (when you are clicking on 'No'), and the second one checks if 'No' is already in the formarray (when you click a regular one, and have No selected)

Reply:
You are totally right, makes sense! Was forgetting the other way around!